File: 2_convert_move/src/main/java/dojo/GameRenderer.java
package dojo;

public class GameRenderer {
    private Game game;

    public GameRenderer(Game game) {
        this.game = game;
    }

    public String render() {
        String message;
        if (game.isGameADraw()) {
            message = "Game is a draw!";
        } else {
            message = "\n" +
                    new BoardRenderer(game.board()).render() + "\n" +
                    "Player " + game.nextCellToPlace() + " please make a move: ";
        }
        return message;
    }
}

File: 8_convert_console_ui/src/main/java/dojo/GameRenderer.kt
package dojo

class GameRenderer(private val game: Game) {

    fun render() =
        if (game.isGameADraw) {
            "Game is a draw!"
        } else {
            "\n" +
                BoardRenderer(game.board()).render() + "\n" +
                "Player " + game.nextCellToPlace() + " please make a move: "
        }
}
